nixos/modules/services/backup/sanoid.nix

```diff
@@ -176,9 +157,13 @@ in {
     # Implementation
 
     config = mkIf cfg.enable {
-      services.sanoid.settings = mkMerge [
-        (mapAttrs' (d: v: nameValuePair ("template_" + d) v.settings) cfg.templates)
-        (mapAttrs (d: v: v.settings) cfg.datasets)
+      services.sanoid.settings =
+        let mergeSettings = v: mkMerge [
+          (lib.mapAttrs (n: lib.mkDefault) (lib.filterAttrs (n: v: n!="settings") v))
+          v.settings
+        ]; in mkMerge [
```

Comment:
Sorry for neglecting this PR, but I think this should be done through the dataset/template specific settings option.

Reply:
Indeed, you're right, sorry. Is it better know?